### PR TITLE
using `# needs` in posets folder

### PR DESCRIPTION
**Pull request by fchapoton**
using the `needs` directive in `posets` folder

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/posets/posets.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+# sage.doctest: needs sage.graphs sage.modules sage.plot sage.rings.finite_rings
```
Comment (mkoeppe) [inline src/sage/combinat/posets/posets.py]: I'd suggest to use the file-level tag `sage.plot` only for files that are focused on plotting. Using block tags for the doctests of plotting is better.
Also sage.rings.finite_rings is actually not needed any more for the doctests in this file, see changes to this file in #35095

**Comment by fchapoton** [comment]
better now ?

File: src/sage/combinat/posets/posets.py [hunk 2]
```diff
@@ -433,15 +434,16 @@ def Poset(data=None, element_labels=None, cover_relations=False, linear_extensio
        With a function that identifies the cover relations: the set
        partitions of `\{1, 2, 3\}` ordered by refinement::
 
-          sage: elms = SetPartitions(3)                                                 # optional - sage.combinat
+          sage: # optional - sage.combinat
```
Comment (mkoeppe) [inline src/sage/combinat/posets/posets.py]: This should be `# needs sage.combinat`

File: src/sage/combinat/posets/posets.py [hunk 3]
```diff
@@ -1931,37 +1933,41 @@ def plot(self, label_elements=True, element_labels=None,
 
         This function can be used without any parameters::
 
+            sage: # optional - sage.plot
```
Comment (mkoeppe) [inline src/sage/combinat/posets/posets.py]: `# needs`

File: src/sage/combinat/posets/posets.py (hunk 2 above)
Comment (mkoeppe) [reply src/sage/combinat/posets/posets.py]: (Just run `./sage -fixdoctests --only-tags` on it to fix)

File: src/sage/combinat/posets/posets.py [hunk 4]
```diff
@@ -3723,18 +3735,18 @@ def magnitude(self) -> Integer:
 
             sage: P1 = posets.RandomPoset(20, 0.05)
```
Comment (mkoeppe) [inline src/sage/combinat/posets/posets.py:3738]: Could use a block tag `sage: # needs sage.libs.flint` here

File: src/sage/combinat/posets/posets.py [hunk 5]
```diff
@@ -8375,17 +8390,17 @@ def greene_shape(self):
         EXAMPLES::
 
             sage: P = Poset([[3,2,1], [[3,1],[2,1]]])
```
Comment (mkoeppe) [inline src/sage/combinat/posets/posets.py:8394]: Could use a block tag here

**Comment by mkoeppe** [comment]
```
File "sage/combinat/posets/posets.py", line 3719, in sage.combinat.posets.posets.FinitePoset.Integer
Failed example:
    needs sage.groups sage.libs.flint
Exception raised:
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1142, in compile_and_execute
        compiled = compiler(example)
      File "/sage/src/sage/doctest/forker.py", line 676, in compiler
        code = compile(example.source, filename, "single",
      File "<doctest sage.combinat.posets.posets.FinitePoset.Integer[0]>", line 1
        needs sage.groups sage.libs.flint
              ^
    SyntaxError: invalid syntax
**********************************************************************
File "sage/combinat/posets/posets.py", line [37](https://github.com/sagemath/sage/actions/runs/5978963391/job/16222089762?pr=36114#step:14:38)37, in sage.combinat.posets.posets.FinitePoset.Integer
Failed example:
    needs sage.groups sage.libs.flint
Exception raised:
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1142, in compile_and_execute
        compiled = compiler(example)
      File "/sage/src/sage/doctest/forker.py", line 676, in compiler
        code = compile(example.source, filename, "single",
      File "<doctest sage.combinat.posets.posets.FinitePoset.Integer[8]>", line 1
        needs sage.groups sage.libs.flint
              ^
    SyntaxError: invalid syntax
```

needs needs #

**Review by mkoeppe — APPROVED** [review]
LGTM. Thanks

**Pull request merged**